src/utils/util/resolver-pipeline/oscal-profile-resolve-merge.xsl

```diff
@@ -77,83 +90,93 @@
         </catalog>
     </xsl:template>
 
-    <xsl:template name="combine-back-matter">
-        <xsl:variable name="here" select="self::catalog"/>
-        <xsl:for-each-group select="back-matter/* | selection/back-matter/*" group-by="true()">
+    <xsl:template name="detect-multiple-structuring-directives" as="empty-sequence()">
+        <xsl:context-item as="element(catalog)" use="required"/>
+        <xsl:variable name="flat" as="element(flat)*" select="merge/flat[.=$true-content]"/>
+        <xsl:variable name="as-is" as="element(as-is)*" select="merge/as-is[.=$true-content]"/>
+        <xsl:variable name="custom" as="element(custom)*" select="merge/custom"/>
+        <xsl:if test="count($flat) + count($as-is) + count($custom) gt 1">
+            <xsl:call-template name="mh:message-handler">
+                <xsl:with-param name="text">Found multiple structuring directives. Choose at most one from: flat (default), as-is, custom.</xsl:with-param>
+                <xsl:with-param name="message-type">Error</xsl:with-param>
+                <xsl:with-param name="terminate" select="true()"/>
+            </xsl:call-template>
+        </xsl:if>
+    </xsl:template>
+
+    <xsl:template name="combine-back-matter" as="element(back-matter)?">
+        <xsl:context-item as="element(catalog)" use="required"/>
+        <xsl:where-populated>
             <back-matter>
                 <!-- Using combination logic on back matter elements. -->
-                <xsl:for-each-group select="current-group()" group-by="(@opr:id,@id,generate-id())[1]">
+                <xsl:for-each-group select="back-matter/* | selection/back-matter/*" group-by="(@opr:id,@uuid,generate-id())[1]">
```

Comment:
Change in group-by: Current schema says the attribute on resource is `@uuid` rather than `@id`.

The other changes on this line relate to use of xsl:where-populated on line 109, which I just thought was more intuitive than <xsl:for-each-group group-by="true()" ...> on line 82 of the old revision. (The GitHub diff is a little hard to read because of the new detect-multiple-structuring-directives template in the middle.)